hibernate-core/src/main/java/org/hibernate/dialect/MariaDBDialect.java

```diff
@@ -80,7 +80,7 @@ public SequenceSupport getSequenceSupport() {
 	@Override
 	public String getQuerySequencesString() {
 		return getSequenceSupport().supportsSequences()
-				? "select table_name from information_schema.TABLES where table_type='SEQUENCE'"
+				? "select table_name from information_schema.TABLES where table_schema = database() and table_type='SEQUENCE'"
```

Comment:
After second thought, I think the SQL statement above might only apply for 10.3 and later for sequence was only introduced since 10.3: https://mariadb.com/kb/en/create-sequence/

For early versions of Mariadb, not sure whether this is correct, however. Could we rely on db version to make the decision? I think this is an example why previous version specific dialect makes sense to some extent, IMHO.